framework/src/meshgenerators/TransfiniteMeshGenerator.C

```diff
@@ -0,0 +1,542 @@
+    break;
+    case 4:
+    {
+      edge = getParsedEdge(parameter, param_vec);
+      mooseAssert(MooseUtils::absoluteFuzzyEqual((edge[0] - P1).norm(), 0.0),
+                  "The parametrization does not fit the first vertex on the edge.");
+      mooseAssert(MooseUtils::absoluteFuzzyEqual((edge[np - 1] - P2).norm(), 0.0),
+                  "The parametrization does not fit the end vertex on the edge.");
+    }
+    break;
+  }
+  if (edge.size() != np)
+    mooseError("The generated edge does not match the number of points on the"
+               "opposite edge.");
+  return edge;
+}
+
+std::vector<Point>
+TransfiniteMeshGenerator::getLineEdge(const Point & P1,
+                                      const Point & P2,
+                                      const std::vector<Real> & param_vec)
+{
+  std::vector<Point> edge;
+  edge.resize(param_vec.size());
+  auto it = 0;
+
+  for (auto rx : param_vec)
+  {
+    edge[it] = P1 * (1.0 - rx) + P2 * rx;
+    it++;
+  };
+  return edge;
+}
+
+std::vector<Point>
+TransfiniteMeshGenerator::getParsedEdge(const std::string & parameter,
+                                        const std::vector<Real> & param_vec)
+{
+  std::vector<Point> edge;
+  Real x_coord, y_coord;
+  edge.resize(param_vec.size());
+
+  std::vector<std::string> param_coords;
+  MooseUtils::tokenize(parameter, param_coords, 1, "&&");
+
+  auto it = 0;
+  for (auto rx : param_vec)
+  {
+    _parsed_func->Parse(param_coords[0], "r");
+    x_coord = _parsed_func->Eval(&rx);
+    _parsed_func->Parse(param_coords[1], "r");
+    y_coord = _parsed_func->Eval(&rx);
+    edge[it] = Point(x_coord, y_coord, 0.0);
+    it++;
+  }
+
+  return edge;
+}
+
+std::vector<Point>
+TransfiniteMeshGenerator::getDiscreteEdge(const unsigned int & np, const std::string & parameter)
+{
+  std::vector<Point> edge;
+
+  std::vector<std::string> string_points;
+  MooseUtils::tokenize(parameter, string_points, 1, "\n");
+  if (string_points.size() != np)
+    mooseError("DISCRETE: the number of discrete points does not match the number of points on the"
+               "opposite edge.");
+
+  edge.resize(np);
+  auto it = 0;
+  for (unsigned int iter = 0; iter < string_points.size(); iter++)
+  {
+    std::vector<Real> point_vals;
+    MooseUtils::tokenizeAndConvert(string_points[iter], point_vals, " ");
+    edge[it] = Point(point_vals[0], point_vals[1], point_vals[2]);
+    it++;
+  }
+
+  return edge;
+}
+
+std::vector<Point>
+TransfiniteMeshGenerator::getCircarcEdge(const Point & P1,
+                                         const Point & P2,
+                                         const std::string & parameter,
+                                         const Point & outward,
+                                         const std::vector<Real> & param_vec)
+{
+  std::vector<Point> edge; // output, to be returned variable
+  edge.resize(param_vec.size());
+
+  std::vector<Real> param_coords;
+  MooseUtils::tokenizeAndConvert(parameter, param_coords, "&&");
```

Comment:
Since we're just getting point coordinates here, should we use whitespace to delimit as we do elsewhere?

Reply:
As per other comments I switched to using semicolons to have consistency from a user perspective.

Reply:
Wait, wait, no you didn't.  You're tokenizing on semicolons in the other parameter parsing, but it is still `&&` here.  *That* explains why that input file which still has `&&` isn't failing.